test/bench/bench.cpp

```diff
@@ -88,72 +88,92 @@ struct ExecutionBenchmarkCase
+        return false;
     }
 
     // Execute once and check results against expectations.
     const auto result = engine.execute(*func_ref, benchmark_case.func_args);
     if (result.trapped)
-        return state.SkipWithError("Trapped");
+    {
+        state.SkipWithError("Trapped");
+        return false;
+    }
 
     if (benchmark_case.expected_result)
     {
         if (!result.value)
         {
             const auto error_msg = "Missing result value, expected: " +
                                    std::to_string(*benchmark_case.expected_result);
-            return state.SkipWithError(error_msg.c_str());
+            state.SkipWithError(error_msg.c_str());
+            return false;
         }
         else if (*result.value != *benchmark_case.expected_result)
         {
             const auto error_msg = "Incorrect result value, expected: " +
                                    std::to_string(*benchmark_case.expected_result) +
                                    ", got: " + std::to_string(*result.value);
-            return state.SkipWithError(error_msg.c_str());
+            state.SkipWithError(error_msg.c_str());
+            return false;
         }
     }
     else if (result.value)
     {
-        return state.SkipWithError(
-            ("Unexpected result value: " + std::to_string(*result.value)).c_str());
+        state.SkipWithError(("Unexpected result value: " + std::to_string(*result.value)).c_str());
+        return false;
     }
     const auto memory = engine.get_memory();
     if (memory.size() < benchmark_case.expected_memory.size())
-        return state.SkipWithError("Result memory is shorter than expected");
+    {
+        state.SkipWithError("Result memory is shorter than expected");
+        return false;
+    }
 
     // Compare _beginning_ segment of the memory with expected.
     // Specifying expected full memory pages is impractical.
     if (!std::equal(std::begin(benchmark_case.expected_memory),
             std::end(benchmark_case.expected_memory), std::begin(memory)))
-        return state.SkipWithError("Incorrect result memory");
+    {
+        state.SkipWithError("Incorrect result memory");
+        return false;
+    }
+
+    return true;
 }
 
 void benchmark_execute(
     benchmark::State& state, EngineCreateFn create_fn, const ExecutionBenchmarkCase& benchmark_case)
 {
-    const auto engine = create_fn();
+    const auto has_memory = !benchmark_case.memory.empty();
 
-    validate_benchmark_case(state, *engine, benchmark_case);
+    const auto engine = create_fn();
+    const bool ok = validate_benchmark_case(state, *engine, benchmark_case);
 
-    const auto has_memory = !benchmark_case.memory.empty();
-    engine->instantiate(*benchmark_case.wasm_binary);
-    const auto func_ref = engine->find_function(benchmark_case.func_name, benchmark_case.func_sig);
+    std::optional<fizzy::test::WasmEngine::FuncRef> func_ref;
+    if (ok)
```

Comment:
Can you explain the changes here in this function?
It looks like when validation is not ok, you would still go and dereference `func_ref` at line 185

Reply:
Ah I guess it works because `validate_benchmark_case` would calll `state.SkipWithError` and the loop below  will not be executed.

But why then not just early return
```
  if (!validate_benchmark_case(...))
    return;
```

Reply:
Yes. The loop will not run, but we need to call `auto _ : state` anyway (this is a libbenchmark limitation in the version we use).

Reply:
Perhaps it's woth to add a comment that we need to call it even when validation failed.